File: Frontend/docs/send.py
#!/usr/bin/python

import os, cgi, smtplib, sys
from email.MIMEMultipart import MIMEMultipart
from email.MIMEBase import MIMEBase
from email.MIMEText import MIMEText
from email import Encoders

print 'Content-Type: text/html\r\n\r\n',


def envio(msg):
	p=os.popen("/opt/haiti/servers/postfix/usr/sbin/sendmail -t -i","w")
	p.write(msg)
	status = p.close()
	if status:
		print "Sendmail exit status", status

##Envio de mail##
def envioMail(toMail, fromMail, asunto, text, attach = ''):
	msg = MIMEMultipart()

	msg['From'] = fromMail 
	msg['To'] = toMail
	msg['Subject'] = asunto 
	msg.attach(MIMEText(text,'html'))

	if attach != '':	
		try:
			part = MIMEBase('application', 'octet-stream')
			part.set_payload(open(attach, 'rb').read())
			Encoders.encode_base64(part)
			part.add_header('Content-Disposition','attachment; filename="%s"' % os.path.basename(attach))
			msg.attach(part)
			
		except:
			envio(msg.as_string())
		envio(msg.as_string())
	else:
		envio(msg.as_string())


diccionario = {}
def parser(cadena):
        global diccionario
        lista = cadena.split("#")
        contador = 1
        while (contador < len(lista)):
                cad = lista[contador]
                claveValor = cad.split("=")
                diccionario[claveValor[0]] = claveValor[1]
                contador = contador + 1
        return diccionario 


def subirTemporal(nombreCampo):
	item = ''
	try:
		item = datos[nombreCampo]
	except:
		pass

	try:
		if item.file:
    			data = item.file.read()
			nombreArchivo = item.filename
			fileHandle = open(nombreArchivo, 'w+')
			fileHandle.write(data)
			fileHandle.close()
			if(os.path.getsize(nombreArchivo) <= 2000000):
				return nombreArchivo
			else:
				os.unlink(nombreArchivo)
				nombreArchivo = ''
				return nombreArchivo
	except:
		return ''


def isTrue(x):
    if x: 
        return 0 
    return 1



adjunto = ''
cuerpoMail = ''
def generarCuerpoDelMail(lineaConf):
	global diccioario
	global cuerpoMail
	parser(lineaConf)
	tipo = diccionario['tipo']
	titulo = diccionario['titulo']
	nombre = diccionario['nombre']
	contenido = datos.getvalue(nombre)
	if isTrue(contenido):
		contenido = 'Sin completar'
	if tipo != 'attach':
		lineaMail = '<tr><td>%s:</td>' % titulo.replace('<','&lt;').replace('>','&gt;')
		lineaMail += '<td>%s</td></tr>\n' % str(contenido.replace('<','&lt;').replace('>','&gt;'))
		cuerpoMail = cuerpoMail + lineaMail
		return True
	else:
		global adjunto
		adjunto = subirTemporal(nombre)
	diccioario = {}


def tw_get_domain():
        linea =  os.environ['SCRIPT_FILENAME']
        lista = linea.split('/')
        if(lista[3] == 'nt'):
                return lista[4] 
        else:
                return lista[3]


def crearEncabezado(cantidad, lineasConf):
	global mailTo, mailFrom, asunto, cuerpoMail
	lineaE = lineasConf[cantidad -1]
	parser(lineaE)
        mailTo = diccionario['mailTo']
        mailFrom = diccionario['mailFrom']
        asunto = diccionario['asunto']
	cuerpoMail = '<table><tr><td colspan=2><b>%s</b></td></tr>' % diccionario['titu']


##Usa el archivo de configuracion para obtener los campos del Formulario y sus nombres##
def obtenerCampos():
        global cuerpoMail
	archivoConfig = tw_get_domain() + '.ini'
        try:
                config = open(archivoConfig,'rt')
        except:
                print "Error al abrir archivo de configuracion: [%s]" % archivoConfig
                sys.exit(1)
        lineas = config.readlines()
        crearEncabezado(len(lineas),lineas)
	contador = 1 
	for linea in lineas:
		if(contador < len(lineas)-1 ):
			generarCuerpoDelMail(linea)
                else:
			if (contador == len(lineas) -1):
				generarCuerpoDelMail(linea)
				cuerpoMail += '</table>'
		contador = contador + 1





######MAIN#######
datos = cgi.FieldStorage()

obtenerCampos()


envioMail(mailTo, mailFrom, asunto, cuerpoMail, adjunto)

if adjunto != '':
	os.unlink(adjunto)

print '<META HTTP-EQUIV="Refresh" CONTENT="0; URL=%s">' % (diccionario['agrade']) 

File: Frontend/docs/send.php
<?
  $adjunto = '';
  $cuerpoMail = '';
  $diccionario = Array();

  function subirTemporal($nombreCampo){
    if((@isset($_POST[$nombre])) && (!@empty($_POST[$nombre]))){
      $nombreArchivo = $_FILES[$nombreCampo]["name"];
      $tamanioArchivo = $_FILES[$nombreCampo]["size"];
      $item = $_FILES[$nombreCampo]["name"];

      if($tamanioArchivo <= 2000000){
        move_uploaded_file($_FILES[$nombreCampo]["tmp_name"], $_FILES[$nombreCampo]["name"]);
        
        return $nombreArchivo;
      }
    }

    return '';
  }


  function generarCuerpoDelMail($lineaConf){
    global $diccionario, $cuerpoMail, $adjunto;

    parser($lineaConf);

    $tipo = $diccionario['tipo'];
    $titulo = $diccionario['titulo'];
    $nombre = $diccionario['nombre'];
   
    if((!@isset($_POST[$nombre])) || (@empty($_POST[$nombre]))){
      $contenido = "Sin completar";
    }else{
      $contenido = $_POST[$nombre];
    }

    if($tipo != 'attach'){
      $lineaMail  = sprintf('<tr><td>%s:</td>', str_replace('>','&gt;', str_replace('<','&lt;', $titulo)));
      $lineaMail .= sprintf('<td>%s</td></tr>', str_replace('>','&gt;', str_replace('<','&lt;', $contenido)));
      $cuerpoMail = $cuerpoMail . $lineaMail;

      return TRUE;
    }else{
      $adjunto = subirTemporal($nombre);
    }

    $diccionario = Array();
  }



  function parser($cadena){
    global $diccionario;

    $lista = explode("#", $cadena);

    $contador = 0;

    while($contador < count($lista)){
      $cad = $lista[$contador];

      $claveValor = explode("=", $cad);

      $diccionario[$claveValor[0]] = $claveValor[1];

      $contador++;
    }

    return $diccionario;
  }


  function crearEncabezado($cantidad, $lineasConf){
    global $mailTo, $mailFrom, $asunto, $cuerpoMail, $diccionario;

    $lineaE = $lineasConf[$cantidad -1];

    parser($lineaE);

    $mailTo = $diccionario['mailTo'];
    $mailFrom = $diccionario['mailFrom'];
    $asunto  = $diccionario['asunto'];

    $cuerpoMail = '<table><tr><td colspan=2><b>' . $diccionario['titu'] . '</b></td></tr>';
  }


  function tw_get_domain(){
    return str_replace('www.', '', $_SERVER['HTTP_HOST']);
  }




  function obtenerCampos(){
    global $cuerpoMail;

    $archivoConfig = tw_get_domain() . '.ini';

    $config = @file_get_contents($archivoConfig);

    if($config === FALSE){
      echo "Error al abrir archivo de configuracion: [" . $archivoConfig . "]";
      exit;
    }

    $lineas = explode("\n", $config);

    crearEncabezado(count($lineas), $lineas);


    for($contador = 1 ; $contador < count($lineas) ; $contador++){
      generarCuerpoDelMail($lineas[$contador]);

      if($contador == count($lineas) - 1){
        $cuerpoMail .= '</table>';
      }
    }
  }

  function envioMail($toMail, $fromMail, $asunto, $text, $attach=''){
    $headers = "";

    $headers .= "From: " . $fromMail; 
    $random_hash = md5(date('r', time())); 

    if($attach != ''){
      $headers .= "\r\nContent-Type: multipart/mixed; boundary=\"PHP-mixed-" . $random_hash . "\""; 

      $attachment = chunk_split(base64_encode(file_get_contents($attach)));

      $message = "--PHP-mixed-" . $random_hash . "
Content-Type: multipart/alternative; boundary=\"PHP-alt-" . $random_hash . "\" 

--PHP-alt-" . $random_hash . "
Content-Type: text/html; charset=\"iso-8859-1\" 
Content-Transfer-Encoding: 7bit

" . $text . "

--PHP-alt-" . $random_hash . "--


--PHP-mixed-" . $random_hash . "
Content-Type: application/zip; name=\"" .$attach ."\"  
Content-Transfer-Encoding: base64  
Content-Disposition: attachment  

" . $attachment . "
--PHP-mixed-" . $random_hash . "-- 
";

    }else{
      $headers .= "\r\nContent-Type: multipart/mixed; boundary=\"PHP-mixed-" . $random_hash . "\""; 

      $message = "--PHP-mixed-" . $random_hash . "
Content-Type: text/html; charset=\"iso-8859-1\" 
Content-Transfer-Encoding: 7bit

" . $text . "

--PHP-mixed-" . $random_hash . "--
";

    }

    mail($toMail, $asunto, $message, $headers);
  }

  obtenerCampos();

  envioMail($mailTo, $mailFrom, $asunto, $cuerpoMail, $adjunto);

  if($adjunto != ''){
    unlink($adjunto);
  }

  echo sprintf('<META HTTP-EQUIV="Refresh" CONTENT="0; URL=%s">', $diccionario['agrade'])
?>